shuffle/cli/src/test.rs

```diff
@@ -66,20 +66,38 @@ async fn create_account(
     root_key_path: &Path,
     client: &DevApiClient,
     factory: &TransactionFactory,
+    network: &Network,
 ) -> Result<(Ed25519PrivateKey, LocalAccount)> {
-    let mut treasury_account = account::get_treasury_account(client, root_key_path).await?;
     let account_key = generate_key::generate_key();
+    let new_account = generate_new_account(&account_key, client).await?;
+    match network.get_faucet_url() {
+        Some(_) => account::create_account_via_faucet(network, &new_account).await?,
+        None => {
+            let mut treasury_account = account::get_treasury_account(client, root_key_path).await?;
+            account::create_account_via_dev_api(
+                &mut treasury_account,
+                &new_account,
+                factory,
+                client,
+            )
+            .await?;
+        }
+    };
+    Ok((account_key, new_account))
+}
+
+async fn generate_new_account(
+    account_key: &Ed25519PrivateKey,
+    client: &DevApiClient,
+) -> Result<LocalAccount> {
     let public_key = account_key.public_key();
     let derived_address = AuthenticationKey::ed25519(&public_key).derived_address();
     let seq_num = client
         .get_account_sequence_number(derived_address)
```

Comment:
You use `generate_new_account` for both faucet and devapi networks, but this requires a `&DevApiClient` to get the sequence number. Thinking out loud, this is because both faucet and devapi networks still expose a devapi for GETs.

An idea might be to pass the seq number in rather than rely on something as heavy as the dev api client, but that's a maybe, and may not be worth the refactor.

Reply:
That's a valid point and an easy refactor. I'll do it!